### PR TITLE
refactor(lint): manage schema files for linter in Deno repo

### PR DESCRIPTION
**Pull request by bartlomieju**
Extracted from https://github.com/denoland/deno/pull/27293.

This commit provides schema files for lint rules and lint tags
in this repo instead of pulling them from `deno_lint` repository.

A unit test was added to ensure all available rules are listed
in the schema file. A unit test for tags can be done once 
https://github.com/denoland/deno/pull/27162 lands.

### CODE REVIEW

File: cli/schemas/config-file.v1.json [hunk 1]
```diff
@@ -300,7 +300,7 @@
               "type": "array",
               "description": "List of rule names that will be excluded from configured tag sets. If the same rule is in `include` it will be run.",
               "items": {
-                "$ref": "https://raw.githubusercontent.com/denoland/deno_lint/main/schemas/rules.v1.json"
+                "$ref": "https://deno.land/x/deno/cli/schemas/lint-rules.v1.json"
```
Comment (bartlomieju) [inline cli/schemas/config-file.v1.json:303]: @nayeemrmn are these correct URLs? Which URL for the the `config-file.v1.json` file does the LSP/vscode use?

Comment (nayeemrmn) [reply cli/schemas/config-file.v1.json:303]: It uses the github URL https://github.com/denoland/vscode_deno/blob/3.42.0/package.json#L596. A relative path like this should work:
```suggestion
                "$ref": "lint-rules.v1.json"
```
If it doesn't work we can verify right after landing and then fix 🤷

Comment (bartlomieju) [reply cli/schemas/config-file.v1.json]: Thanks, let's do that 👍

**Review by nayeemrmn — APPROVED** [review]
LGTM

**Pull request merged**